tamr_unify_client/models/project/resource.py

```diff
@@ -78,6 +78,24 @@ def as_mastering(self):
             )
         return MasteringProject(self.client, self._data, self.api_path)
 
+    def add_source_dataset(self, dataset):
+        """
+        Associate a dataset with a project in Unify.
+
+        By default, datasets are not associated with any projects.
+        They need to be added as input to a project before they can be used
+        as part of that project
+
+        :param project: Unify Project
+        :param dataset: Unify Dataset
+        :return: None
+        """
+        dataset_id = dataset.relative_id.split("/")[-1]
+        self.client.post(
+            self.api_path + "/inputDatasets" + f"?id={dataset_id}"
+        ).successful()
+        return None
```

Comment:
Similarly to how we address #90 , you should opt here to return the response from the server (rather than `None`):

```python
response = self.client.post(
            self.api_path + "/inputDatasets" + f"?id={dataset_id}"
        ).successful()
return response
```